services/galley/src/Galley/API/Teams.hs

```diff
@@ -391,12 +394,19 @@ finishCreateTeam team owner others zcon = do
     push1 $ newPush1 zusr (TeamEvent e) (list1 (userRecipient zusr) r) & pushConn .~ zcon
     pure (empty & setStatus status201 . location (team^.teamId))
 
-getBindingTeamMembers :: UserId -> Galley Response
-getBindingTeamMembers zusr = do
+withBindingTeam :: UserId -> (TeamId -> Galley b) -> Galley b
+withBindingTeam zusr callback = do
     tid <- Data.oneUserTeam zusr >>= ifNothing teamNotFound
     binding <- Data.teamBinding tid >>= ifNothing teamNotFound
     case binding of
-        Binding -> do
-            members <- Data.teamMembers tid
-            pure $ json $ teamMemberListJson True (newTeamMemberList members)
+        Binding -> callback tid
         NonBinding -> throwM nonBindingTeam
+
+getBindingTeamId :: UserId -> Galley Response
+getBindingTeamId zusr = withBindingTeam zusr $ \tid -> pure $ json tid
```

Comment:
This could be shortened to: `getBindingTeamId zusr = withBindingTeam zusr $ pure . json`